lib/airbrake/rack/instrumentable.rb

```diff
@@ -14,15 +16,62 @@ module Rack
     # @api public
     # @since v9.2.0
     module Instrumentable
+      # rubocop:disable Metrics/AbcSize
+      # rubocop:disable Metrics/MethodLength
+      # rubocop:disable Metrics/CyclomaticComplexity
+      # rubocop:disable Metrics/PerceivedComplexity
```

Comment:
The only reason this code is so complex is because it needs to be to handle the various problems that come up in this situation. I can move the code to self methods on the module so they don't pollute the namespace of the classes that include this module, and that will remove the warnings, but what I can't do is make it less complex for real.